File: backend/src/main/java/com/musicmatch/backend/dto/ProfileCardDTO.java
package com.musicmatch.backend.dto;

import java.util.List;

import com.musicmatch.backend.model.ExperienceLevel;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class ProfileCardDTO {

    private Long id;
    private String username;
    private String bio;
    private String city;
    private String profilePicture;
    private List<Long> styles;
    private List<InstrumentLevelResponse> instruments;
    private double score;
    private ExperienceLevel profileLevel;
}

File: mobile/app/src/main/java/com/musicmatch/mobile/model/dto/ProfileCardDTO.kt
package com.musicmatch.mobile.model.dto

import com.musicmatch.mobile.model.ExperienceLevel

data class ProfileCardDTO(
    val id: Long,
    val username: String,
    val bio: String?,
    val city: String?,
    val profilePicture: String?,
    val styles: List<Long>?,
    val instruments: List<InstrumentLevelResponse>,
    val score: Double,
    val profileLevel: ExperienceLevel
)